fix DietPi-Set_hardware > lcdpanel   (waveshare32)

guiksign opened the pull request:
curl command will fail if destination folder does not exist

<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->

Note, this fix was written from master branch but no diff for that file with dev branch

guiksign committed:
fix DietPi-Set_hardware > lcdpanel   (waveshare32)

curl command will fail if destination folder does not exist

guiksign committed:
fix DietPi-Set_hardware > lcdpanel   (waveshare32)

curl command will fail if destination folder does not exist

guiksign committed:
update to pass shell check

guiksign committed:
update to pass shell check

guiksign committed:
damn, this is really picky....

guiksign committed:
damn, this is really picky....

MichaIng commented on dietpi/func/dietpi-set_hardware:
```diff
@@ -723,6 +723,9 @@ _EOF_
+			# Check if destination folder overlays exist before running curl command
+			[[ -d '/boot/overlays' ]] || G_EXEC mkdir -p /boot/overlays
+
 			G_EXEC curl -sSfL 'https://raw.githubusercontent.com/waveshare/LCD-show/master/waveshare32b-overlay.dtb' -o /boot/overlays/waveshare32b.dtbo
```
Many thanks. With the new firmware stack, the directory needs to be `/boot/firmware/overlays`.
```suggestion
			# Download overlay to correct dir based on legacy vs modern firmware
			local fp_overlays='/boot/firmware/overlays'
			dpkg-query -s 'raspi-firmware' &> /dev/null || fp_overlays='/boot/overlays'
			G_EXEC curl -sSfL 'https://raw.githubusercontent.com/waveshare/LCD-show/master/waveshare32b-overlay.dtb' -o "$fp_overlays/waveshare32b.dtbo"
```

guiksign committed:
suggestion from MichaIng to use $fp_overlays

Co-authored-by: MichaIng <micha@dietpi.com>

guiksign committed:
suggested change + fix

guiksign committed:
fixing static code analysis

guiksign committed:
yes ok, I am old, I wear glasses...

guiksign committed:
a year later

MichaIng commented on dietpi/func/dietpi-set_hardware:
```diff
@@ -723,7 +723,14 @@ _EOF_
+			dpkg-query -s 'raspi-firmware' &> /dev/null || fp_overlays='/boot/overlays'
+
+			# Check if destination folder overlays exist before running curl command
+ 			[[ -d "$fp_overlays" ]] || G_EXEC mkdir -p "$fp_overlays"
```
Actually, if this directory does not exist, there is no kernel package installed, or at least not the expected one. In that case, we better error out. So I would skip that.

The problem in your case (and for everyone with a newer RPi image) is the change boot mount point from `/boot` to `/boot/firmware`.

MichaIng committed:
dietpi-set_hardware: fail on nonexisting overlay dir

as it is expected part of the RPi kernel package.

Remove the waveshare32 overlay as well from new overlays dir location.
